src/app/clientPage.tsx

```diff
@@ -0,0 +1,118 @@
+"use client";
+import { useState } from "react";
+import InputComponent from "~/components/Input/Input";
+import { InputFormSubmissionType, ResponseType, SongMeaningResponseType, TextSummaryResponseType } from "~/types";
+import InputPageHeader from "~/components/Input/InputPageHeader";
+import Result from "~/components/Result/Result";
+import Loading from "~/components/Loading/Loading";
+import Error from "~/components/Error/Error";
+import useOpenAiSSEResponse from "~/hooks/useOpenAiSSEResponse";
+import useAnalytics from "~/hooks/use-analytics";
+
+export default function ClientPage({ searchParams }: { searchParams?: { [key: string]: string | undefined } }) {
+  const [originalContent, setOriginalContent] = useState(searchParams?.original ?? "");
+  const [displayResult, setDisplayResult] = useState(
+    searchParams?.original &&
+      searchParams?.result &&
+      searchParams?.original?.length > 0 &&
+      searchParams?.result?.length > 0,
+  );
+  const [currentResult, setCurrentResult] = useState<ResponseType | null>(
+    searchParams?.result && JSON.parse((searchParams.result as string) ?? null),
```

Comment:
I believe query param values can be string, undefined, or array. We have a small util to get the first string of an array. You could try using that to get the correct type here. :)

`getStringOrFirst()`

Reply:
oh yeah i remember you added that a while back, will use it thanks!